docs/2022-07-18-JSON-RPC-Server-API.md

```diff
@@ -438,6 +438,52 @@ requests to the server by passing `"module": "MODULE-NAME"`.
 }
 ```
 
+## Get-model
+
+A `get-model` request aims to find a variable substitution that satisfies all predicates in the provided kore term, or else responds that it is not satisfiable. The request may also fail to obtain an answer.
+
+### Request:
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "method": "get-model",
+  "params": {
+    "state": {"format": "KORE", "version": 1, "term": {...}},
+    "module": "MODULE-NAME"
+  }
+}
+```
+
+Optional parameters: `module` (main module name)
+
+### Error Response:
+
+same as for `execute`
+
+### Correct Response:
+
+```json
+{
+  "jsonrpc":"2.0",
+  "id":1,
+  "result":{
+    "satisfiable": "is-satisfiable",
+    "substitution": {"format": "KORE", "version": 1, "term": {...}}
```

Comment:
Can this field be more structured, i.e. a JSON with variable names as keys and terms as values?

Also, I think for the whole API we should drop the

```json
{"format": "KORE", "version": 1, "term": {...}}
```

structure and just use

```json
{...}
```

for terms. For a given version of the API we should never encounter a situation where one KORE term is version `1` and another term is version `2`, thus the KORE format version is implied by the API version. So even if want to be extra prudent about API versioning, we can either put the version in the `result` root once, or simply add a `version` endpoint (but I think we should be fine without either, integration tests are expected to catch these kind of bugs).

Reply:
Well... I agree it seems superfluous, but I added that version and format because it was requested when I first made the json format.
I'd rather stick with it now than change loads of code in various places to remove format and version.

Reply:
As for the substitution structure, we have the equation encoding for substitutions in other places, too, and it has the advantage that field names will always be known statically (which I prefer by a great deal). It is an easy transformation to convert to a dictionary from the equations.

Reply:
> I'd rather stick with it now

Sure, it's not something to address in this PR. But if you agree it's a change that would make the API better, we can track it in an issue.

> field names will always be known statically

How about this format then:

```json
[{"var:" "X", "term": {...}}, {"var": "Y", "term": {...}}]
```

(`"var"` can also be an `EVar` JSON.)

I think it's better to have this parsed format as it gives stronger static guarantees on the structure.

If you'd stick with the equation format for now for consistency, that's fine. But if this is something that you'd also consider an improvement, let's make an issue for it.